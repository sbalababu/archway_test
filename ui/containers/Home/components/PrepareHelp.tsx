import * as React from 'react';
import SyntaxHighlighter from 'react-syntax-highlighter';
import { tomorrowNightEighties } from 'react-syntax-highlighter/dist/styles/hljs';
import Label from './Label';

interface Props {
  location: string;
  namespace: string;
}

const SetupHelp = ({ location, namespace }: Props) => (
  <div style={{ padding: '4px' }}>
    <Label>step 1: prepare</Label>
    <p
      style={{
        marginBottom: '0.5em',
        fontSize: '12px',
        textAlign: 'center',
      }}
    >
      set up a table for your work in your namespace
    </p>
    <SyntaxHighlighter language="sql" style={{ overflow: 'auto', ...tomorrowNightEighties }}>
      {`CREATE TABLE ${namespace}.new_data_landing
LOCATION '${location}/new_data/landing'`}
    </SyntaxHighlighter>
  </div>
);

export default SetupHelp;
