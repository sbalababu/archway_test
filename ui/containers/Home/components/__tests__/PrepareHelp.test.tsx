import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PrepareHelp from '../PrepareHelp';

describe('PrepareHelp', () => {
  it('renders correctly', () => {
    const props = {
      namespace: 'user_ruud_kaland_rkaland',
      location: 'hdfs://valhalla/user/ruud_kaland_rkaland/db',
    };
    const wrapper = shallow(<PrepareHelp {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
