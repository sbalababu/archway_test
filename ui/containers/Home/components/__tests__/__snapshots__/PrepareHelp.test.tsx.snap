// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrepareHelp renders correctly 1`] = `
<div
  style={
    Object {
      "padding": "4px",
    }
  }
>
  <Label>
    step 1: prepare
  </Label>
  <p
    style={
      Object {
        "fontSize": "12px",
        "marginBottom": "0.5em",
        "textAlign": "center",
      }
    }
  >
    set up a table for your work in your namespace
  </p>
  <SyntaxHighlighter
    language="sql"
    style={
      Object {
        "hljs": Object {
          "background": "#2d2d2d",
          "color": "#cccccc",
          "display": "block",
          "overflowX": "auto",
          "padding": "0.5em",
        },
        "hljs-addition": Object {
          "color": "#99cc99",
        },
        "hljs-attribute": Object {
          "color": "#ffcc66",
        },
        "hljs-built_in": Object {
          "color": "#f99157",
        },
        "hljs-builtin-name": Object {
          "color": "#f99157",
        },
        "hljs-bullet": Object {
          "color": "#99cc99",
        },
        "hljs-comment": Object {
          "color": "#999999",
        },
        "hljs-deletion": Object {
          "color": "#f2777a",
        },
        "hljs-emphasis": Object {
          "fontStyle": "italic",
        },
        "hljs-keyword": Object {
          "color": "#cc99cc",
        },
        "hljs-link": Object {
          "color": "#f99157",
        },
        "hljs-literal": Object {
          "color": "#f99157",
        },
        "hljs-meta": Object {
          "color": "#f99157",
        },
        "hljs-name": Object {
          "color": "#f2777a",
        },
        "hljs-number": Object {
          "color": "#f99157",
        },
        "hljs-params": Object {
          "color": "#f99157",
        },
        "hljs-quote": Object {
          "color": "#999999",
        },
        "hljs-regexp": Object {
          "color": "#f2777a",
        },
        "hljs-section": Object {
          "color": "#6699cc",
        },
        "hljs-selector-class": Object {
          "color": "#f2777a",
        },
        "hljs-selector-id": Object {
          "color": "#f2777a",
        },
        "hljs-selector-tag": Object {
          "color": "#cc99cc",
        },
        "hljs-string": Object {
          "color": "#99cc99",
        },
        "hljs-strong": Object {
          "fontWeight": "bold",
        },
        "hljs-symbol": Object {
          "color": "#99cc99",
        },
        "hljs-tag": Object {
          "color": "#f2777a",
        },
        "hljs-template-variable": Object {
          "color": "#f2777a",
        },
        "hljs-title": Object {
          "color": "#6699cc",
        },
        "hljs-type": Object {
          "color": "#f99157",
        },
        "hljs-variable": Object {
          "color": "#f2777a",
        },
        "overflow": "auto",
      }
    }
  >
    CREATE TABLE user_ruud_kaland_rkaland.new_data_landing
LOCATION 'hdfs://valhalla/user/ruud_kaland_rkaland/db/new_data/landing'
  </SyntaxHighlighter>
</div>
`;
